File: Matrix/Row with Max 1s/main.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function template for C++
class Solution {
  public:
    int rowWithMax1s(vector<vector<int> > &arr) {
        int ans_ind = -1;
        
        int n = arr.size();
        int m = arr[0].size();
        
        int j = m-1;
        
        for(int i = 0;i<n;i++) {
            while(j >= 0 && arr[i][j] == 1) {
                ans_ind = i;
                j--;
            }
        }
        return ans_ind;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, m;
        cin >> n >> m;
        vector<vector<int> > arr(n, vector<int>(m));
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                cin >> arr[i][j];
            }
        }
        Solution ob;
        auto ans = ob.rowWithMax1s(arr);
        cout << ans << "\n";

        cout << "~"
             << "\n";
    }
    return 0;
}

// } Driver Code Ends

File: Matrix/Row with Max 1s/main.py
class Solution:
    def rowWithMax1s(self, arr):
        ans_ind: int = -1
        
        n: int = len(arr)
        m: int = len(arr[0])
        
        j: int = m-1
        
        for i in range(n):
            while(j>=0 and arr[i][j] == 1):
                ans_ind = i
                j-=1

        return ans_ind


#{ 
 # Driver Code Starts
if __name__ == "__main__":
    import sys
    input = sys.stdin.read
    data = input().split()

    idx = 0
    tc = int(data[idx])
    idx += 1

    for _ in range(tc):
        n = int(data[idx])
        m = int(data[idx + 1])
        idx += 2

        arr = []
        for i in range(n):
            arr.append(list(map(int, data[idx:idx + m])))
            idx += m

        ans = Solution().rowWithMax1s(arr)
        print(ans)
        print("~")

# } Driver Code Ends